pkg/agent/agentclient/client.go

```diff
@@ -33,21 +33,33 @@ type AgentClient struct {
 	nextConnID  int64
 	connContext map[int64]*connContext
 
-	stream *RedialableAgentClient
+	stream      *RedialableAgentClient
+	serverID    string // the proxy server this client connects to
+	serverCount int    // number of proxy servers, 1 if server is not HA.
```

Comment:
AgentClient and RedialableAgentClient seem like the wrong level for keeping the serverCount. I realize the value comes from Connect() but I think it makes more sense to plumb the value up to the AgentClientSet.

Reply:
Might make more sense to pass a admin or metadata channel in to the Clients. They can then push things like server count out to the appropriate Set or controller when something like ServerCount is set (changed)

Reply:
> I realize the value comes from Connect() but I think it makes more sense to plumb the value up to the AgentClientSet.

That requires plumbing through 4 functions. And because the server count is only useful to the clientset, we will have many "_, err := xxx()" in our code.

Using a channel is cleaner, and is required to deal with HA proxy servers that can dynamically resize, but also makes the code more complex. As it's not needed yet, I'll keep the current code.